tests/torch/fx/test_compress_weights.py

```diff
@@ -0,0 +1,257 @@
+            spies.append(spy)
+    assert num_decompression_nodes == 2
+
+    compressed_model(input_ids)
+
+    for spy in spies:
+        assert spy.call_count == 1
+
+
+@pytest.mark.parametrize("mode", (CompressWeightsMode.INT8_SYM, CompressWeightsMode.INT8_ASYM))
+def test_compressed_model_inference(mode):
+    torch.manual_seed(42)
+    model = ShortTransformer(5, 10, share_weights=True)
+    input_ids = torch.randint(0, 10, (5,))
+    exported_model = _capture_model(model, input_ids)
+    exported_model_output = exported_model(input_ids)
+    compressed_model = compress_weights(exported_model, mode=mode)
+    compressed_model_outputs = compressed_model(input_ids)
+
+    assert (
+        exported_model_output.shape == compressed_model_outputs.shape
+    ), "Compressed model output shape is not equal to the model output shape"
+    assert torch.all(torch.isclose(exported_model_output, compressed_model_outputs, atol=1)).item()
+
+
+@pytest.mark.parametrize("mode", (CompressWeightsMode.INT8_SYM, CompressWeightsMode.INT8_ASYM))
+def test_compress_weights_model_size_conv(mode):
+
+    dtype = torch.int8 if mode == CompressWeightsMode.INT8_SYM else torch.uint8
+    model = ConvolutionModel()
+
+    input_ids = torch.randint(0, 10, [1, 3, 300, 300])
+    exported_model = _capture_model(model, input_ids)
+    model_size = get_model_size(exported_model)
+    compressed_model = compress_weights(exported_model, mode=mode)
+    compressed_model_size = get_model_size(compressed_model)
+
+    n_compressed_weights = 0
+    n_target_modules = 0
+    compressed_node_weight_port = {"linear": 1, "conv2d": 1, "conv_transpose2d": 1}
+
+    n_target_modules, n_compressed_weights = get_compressed_modules_weights(
+        compressed_model, dtype, compressed_node_weight_port
+    )
+
+    assert n_compressed_weights == n_target_modules
+    assert compressed_model_size < model_size
+
+
+@pytest.mark.parametrize("mode", (CompressWeightsMode.INT8_SYM, CompressWeightsMode.INT8_ASYM))
+def test_compress_weights_functional_model(mode):
+    model = FunctionalModel()
+    decompressor_type = "symmetric" if mode == CompressWeightsMode.INT8_SYM else "asymmetric"
+
+    input_ids = torch.randint(0, 10, [1, 3, 300, 300])
+    exported_model = _capture_model(model, input_ids)
+    compressed_model = compress_weights(exported_model, mode=mode)
+
+    n_compressed_weights = 0
+
+    for node in compressed_model.graph.nodes:
+        if decompressor_type in node.name:
+            n_compressed_weights += 1
+    assert n_compressed_weights == 4
+
+
+@pytest.mark.parametrize("mode", SUPPORTED_MODES)
+@pytest.mark.parametrize(
+    "params",
+    (
+        {"ratio": 0.5},
+        {"group_size": 64},
+        {"all_layers": True},
+        {"all_layers": False},
+        *({"sensitivity_metric": metric} for metric in ALL_SENSITIVITY_METRICS),
+        {"gptq": True},
+        {"awq": True},
+        {"scale_estimation": True},
+        {"lora_correction": True},
```

Comment:
`subset_size` and `dataset` are also not supported.
If https://github.com/openvinotoolkit/nncf/pull/2978 will be merged before this PR, there's also a `backup_precision` parameter.

Reply:
hm, so in the case of `backup_precision`, I can leave it for now depending on if the PR is merged?

Reply:
Following @alexsu52's guidance, the subset_size parameter is ignored when the dataset is None in the WeightsCompression Algorithm. For this reason, I didn't add a check for subset_size, but I did include a check for dataset.